src/components/AudioPlayer/ProgressBar.js

```diff
@@ -37,21 +40,44 @@ class ProgressBar extends TrackPlayer.ProgressComponent {
     TrackPlayer.seekTo(seekValueSec)
   }
 
+  // resets progress to zero and calls on the action for the end of a song
+  endTrack = async () => {
+    const { updatePlayed, updateProgress, topScreen, endSong } = this.props
+    updateProgress(0)
+    updatePlayed(0)
+    await TrackPlayer.seekTo(0)
+
+    if (topScreen) endSong()
+  }
+
   // In the mobile version, played, duration, and progress are all updated by
-  // the progress bar, not the AudioPlayerSub, so if there are differences
+  // the render function in ProgressBar, not the AudioPlayerSub, so if there are differences
   // between the new props and the old props, don't rerender.
   // Prevents an infinite loop.
   shouldComponentUpdate(nextProps) {
-    //played, duration, progress
-    const { played, duration, progress, endSong } = this.props
-    // Check if the song has ended, and if so call the endSong function from index
-    if (Math.round(progress * 100) / 100 === 1) endSong()
+    const {
+      played,
+      duration,
+      progress,
+      topScreen,
+      switching,
```

Comment:
It looks like only `startSwitch` is used, not `switching`. It would be worth clearing this up.

Reply:
Yup! I'll get rid of that since `startSwitch` deals with the renders in `progressBar.js` and `switching` deals with the actual logic of changing tracks in `AudioPlayer.js`